haystack/nodes/base.py

```diff
@@ -207,3 +235,13 @@ def _dispatch_run(self, **kwargs) -> Tuple[Dict, str]:
 
         output["params"] = params
         return output, stream
+
+    @classmethod
+    def _get_signature(cls) -> Dict[str, inspect.Parameter]:
```

Comment:
Just FYI: there's another very similar method in `_json_schema.py`: https://github.com/deepset-ai/haystack/blob/b8a3c930f01f26239897582dc9606a594b4d1735/haystack/nodes/_json_schema.py#L90

It's interesting that you seem to have taken a completely different approach to the problem... I don't think we need any action here though